### ISSUE TITLE
display updates do not work properly in cocalc-jupyter

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/cocalc#7164]
This input
```py
handle = display("before", display_id=True)
handle.update("after")
```
should result in the output `'after'`.  

<img width="1438" alt="image" src="https://github.com/sagemathinc/cocalc/assets/1276278/7d5172e1-e60d-4551-b2bf-73504776d7ef">

---

> @rgbkrk writes: "This should end up as a single output that reads "after". This also should work across cells, defining the handle in a prior cell and updating it later. It's very useful for progress bars without needing widgets as well as updating an image during generation. As an example, OpenAI uses this feature in this cookbook: https://cookbook.openai.com/examples/gpt_with_vision_for_video_understanding"

**Comment by rgbkrk** [comment]
Thanks!

The protocol documentation: https://jupyter-client.readthedocs.io/en/latest/messaging.html#update-display-data

**Comment by williamstein** [comment]
Just a remark that I recently broke this in the opposite direction.  Now there is no output at all:

<img width="1600" alt="image" src="https://github.com/user-attachments/assets/e25bb416-87ec-43e9-bddd-e8ce7c9b6c26">

I broke this because I was reading the spec for other reasons (namely modernizing our ipywidgets support), noticed these new transient messages, read that transient messages are optional, and changed the frontend rendering code to ignore them.  So now nothing is displayed in this case.
